docs/core/tools/create-custom-templates.md

```diff
@@ -0,0 +1,232 @@
+
+## Configuration
+
+A template is composed of the following components:
+
+- Source files and folders
+- Configuration file (*template.json*)
+
+### Source files and folders
+
+The source files and folders include whatever files and folders you want the template engine to use when the `dotnet new <TEMPLATE>` command is executed. The template engine is designed to use *runnable projects* as source code to produce projects. This has several benefits:
+
+- The template engine doesn't require you to inject special tokens into your project's source code.
+- The code files aren't special files or modified in any way to work with the template engine, so the tools you normally use when working with projects also work with template content.
+- You build, run, and debug your template projects just like you do for any of your other projects.
+- You can create a template from an existing project quickly by merely adding a *template.json* configuration file to the project.
+
+Files and folders stored in the template aren't limited to formal .NET project types, such as a .NET Core or .NET Framework solutions. Source files and folders may consist of any content that you wish to create when the template is used, even if the template engine produces just one file for its output, such as a configuration file or a solution file. For example, you can create a template that contains a single *web.config* source file that creates a modified *web.config* file. The modifications to source files are based on logic and settings you've provided in the *template.json* configuration file along with values provided by the user passed as options to the `dotnet new <TEMPLATE>` command.
+
+### template.json
+
+The *template.json* file is placed in a *.template.config* folder in the root directory of the template. The file provides configuration information to the template engine. The minimum configuration requires the members shown in the following table, which is sufficient to add to a .NET Core app to produce a functional template.
+
+| Member            | Type          | Description |
+| ----------------- | ------------- | ----------- |
+| `author`          | string        | The author of the template. |
+| `classifications` | array(string) | Zero or more characteristics of the template that a user might search for it by. |
+| `identity`        | string        | A unique name for this template. |
+| `name`            | string        | The name for the template that users should see. |
+| `shortName`       | string        | A default shorthand for selecting the template (applies to environments where the template name is specified by the user - not selected via a GUI). |
+
+**Example**
+
+```json
+{
+  "author": "Catalina Garcia",
```

Comment:
It would be nice to add

```
"$schema": "http://json.schemastore.org/template",
```

With this included any editor that support JSON Schema will be enabled. For VS Code this is needed to get IntelliSense when editing the `template.json` file.